File: project-euler/0012/0012.js
var sequence = 1;
var triangle = -1;
var divisors = [];

while(divisors.length < 500) {

    triangle = getTriangleNumber(sequence);
    divisors = getDivisors(triangle);
    sequence += 1;
}

console.log(triangle);

function getTriangleNumber(sequence) {
    return ((1 + sequence) * sequence) / 2;
}

function getDivisors(number) {
    var divisors = [];
    var quotient = number;
    var seq = 1;

    while(quotient > seq) {
        if(number % seq === 0) {

            var quotient = number / seq;

            divisors.push(seq);

            if(quotient !== seq) {
                divisors.push(quotient);
            }
        }
        seq += 1;
    }
    return divisors;
}

File: project-euler/0012/0012.pl
#!/usr/bin/env perl

$sequence = 1;
$triangle = -1;
@divisors = ();

while(@divisors + 0 < 500) {
    $triangle = ((1 + $sequence) * $sequence) / 2;
    @divisors = &getDivisors($triangle);
    $sequence += 1;
}

print $triangle;

sub getDivisors {
    my $number = $_[0];
    my @divisors = ();
    my $quotient = $number;
    my $seq = 1;

    while($quotient > $seq) {
        if($number % $seq == 0) {
            $quotient = $number / $seq;
            push(@divisors, $seq);
            push(@divisors, $quotient) if $quotient != $seq;
        }
        $seq += 1;
    }
    return @divisors;
}
